src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1106,3 +1103,140 @@ export const handleEncryptPreviewSubmission: RequestHandler<
+      AuthService.getFormAfterPermissionChecks({
+        user,
+        formId,
+        level: PermissionLevel.Read,
+      }),
+    )
+    .andThen(EmailSubmissionService.checkFormIsEmailMode)
+  if (formResult.isErr()) {
+    logger.error({
+      message: 'Error while retrieving form for preview submission',
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapEmailSubmissionError(
+      formResult.error,
+    )
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+  const form = formResult.value
+
+  const parsedResponsesResult = await EmailSubmissionService.validateAttachments(
+    responses,
+  ).andThen(() => SubmissionService.getProcessedResponses(form, responses))
+  if (parsedResponsesResult.isErr()) {
+    logger.error({
+      message: 'Error while parsing responses for preview submission',
+      meta: logMeta,
+      error: parsedResponsesResult.error,
+    })
+    const { errorMessage, statusCode } = mapEmailSubmissionError(
+      parsedResponsesResult.error,
+    )
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+  const parsedResponses = parsedResponsesResult.value
+  const attachments = mapAttachmentsFromResponses(req.body.responses)
+
+  // Handle SingPass, CorpPass and MyInfo authentication and validation
+  if (form.authType === AuthType.SP || form.authType === AuthType.MyInfo) {
+    parsedResponses.push(
+      ...createSingpassParsedResponses(PREVIEW_SINGPASS_UINFIN),
+    )
+  } else if (form.authType === AuthType.CP) {
+    parsedResponses.push(
+      ...createCorppassParsedResponses(
+        PREVIEW_CORPPASS_UINFIN,
+        PREVIEW_CORPPASS_UID,
+      ),
+    )
+  }
+
+  const emailData = new SubmissionEmailObj(
+    parsedResponses,
+    // All MyInfo fields are verified in preview
+    new Set(AdminFormService.extractMyInfoFieldIds(form.form_fields)),
+    form.authType,
+  )
+  const submission = EmailSubmissionService.createEmailSubmissionWithoutSave(
+    form,
+    // Don't need to care about response hash or salt
+    '',
+    '',
+  )
+
+  const sendAdminEmailResult = await MailService.sendSubmissionToAdmin({
+    replyToEmails: EmailSubmissionService.extractEmailAnswers(parsedResponses),
+    form,
+    submission,
+    attachments,
+    dataCollationData: emailData.dataCollationData,
+    formData: emailData.formData,
+  })
+  if (sendAdminEmailResult.isErr()) {
+    logger.error({
+      message: 'Error sending submission to admin',
+      meta: logMeta,
+      error: sendAdminEmailResult.error,
+    })
+    const { statusCode, errorMessage } = mapEmailSubmissionError(
+      sendAdminEmailResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+
+  // Return the reply early to the submitter
+  res.json({
+    message: 'Form submission successful.',
+    submissionId: submission.id,
+  })
+
+  return SubmissionService.sendEmailConfirmations({
```

Comment:
i know i didn't flag in it in the preview encrypt-submission PR, but should we reverse the order?

don't await the `sendEmailConfirmations` function and return the res.json instead?

Reply:
ah okay, sure we can do that. will change both!